### DISCUSSION TITLE
cause the build to fail on lint errors

### DISCUSSION THREAD

**Discussion by ahl** [11ty/eleventy#4053] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
The linter capability is extremely cool, but I'd ideally like to force the build to be lint clean, i.e. for eleventy to fail if a linter detects a problem.

**Describe the solution you'd like**
Either return codes for linter functions or APIs for linters to invoke e.g.:
```javascript
    eleventyConfig.addLinter("lint", async function(content, inputPath, outputPath) {
    // found a problem
    return { status: "error", msg: "I am sad" };
    // all good
    return { status: "ok" };
  });
```
```javascript
    eleventyConfig.addLinter("lint", async function(content, inputPath, outputPath, lint) {
    lint.warn("you should probably fix this");
    lint.error("THE SKY IS FALLING");
  });
```

**Describe alternatives you've considered**
The linter could just exit the process, but that seems extreme.

**Comment by Ryuno-Ki** [comment]
Oh my! I am imagine moving to eleventy and using the inclusive language linter...

**Comment by zachleat** [comment]
Ah I like this. @Ryuno-Ki not sure why the bad vibes here, it seems like it would need to be an optional feature (not all linter errors should fail)

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by Ryuno-Ki** [comment]
> Ah I like this. @Ryuno-Ki not sure why the bad vibes here

I used the Inclusive Language Filter on downloaded tweets, which I have as part of the build.
It cries out over mentions of https://twitter.com/EasyChirp because it contains „easy”.

Yeah, would be cool if I could tell the linter to skip certain instances / directories (e.g. via frontmatter).

**Comment by NickColley** [comment]
If anyone has figured out a workaround for this, it'd be good, I have an accessibility linter that I only run in CI that I'd like to fail my CI builds. :smile: 

Update: I ended up doing this, which is hacky but works for what I need.

```json
{
    "scripts": {
        "test:ci": "npm run test 2>&1 | tee -a /tmp/ci-output.txt; if grep -Fq 'Linter' /tmp/ci-output.txt; then exit 1; fi",
        "test": "NODE_ENV=test eleventy"
    }
}
```

It runs the regular test, outputting both stdout and stderr to a temporary file. Then uses grep to look for the word 'Linter' and exits with a failing code if it does. You may need to change the word 'Linter' to something you're looking for.

**Comment by Ryuno-Ki** [comment]
FWIW: Normally, a CI (like CircleCI) wouldn't allow you to write to a system environment (I think). Thus, you would use `./tmp/ci-output.txt` there. (When in doubt, run `mkdir -p ./tmp` first to ensure the presence of the directory).

**Comment by NickColley** [comment]
@Ryuno-Ki good note, I think potentially because GitHub Actions is spinning it's own instance of Ubuntu you can use that temp folder.
